x-pack/plugins/security_solution/server/search_strategy/event_filters_fields/index.test.ts

```diff
@@ -0,0 +1,169 @@
+
+describe('Event filters fields', () => {
+  const getFieldsForWildcardMock = jest.fn();
+  const esClientSearchMock = jest.fn();
+  const esClientFieldCapsMock = jest.fn();
+  const endpointAppContextService = createMockEndpointAppContextService();
+  let IndexPatterns: DataViewsServerPluginStart;
+
+  const deps = {
+    esClient: {
+      asInternalUser: { search: esClientSearchMock, fieldCaps: esClientFieldCapsMock },
+    },
+  } as unknown as SearchStrategyDependencies;
+
+  const mockPattern = {
+    title: 'test',
+    fields: {
+      toSpec: () => ({
+        coolio: {
+          name: 'name_test',
+          type: 'type_test',
+          searchable: true,
+          aggregatable: true,
+        },
+      }),
+    },
+    toSpec: () => ({
+      runtimeFieldMap: { runtimeField: { type: 'keyword' } },
+    }),
+  };
+  const getStartServices = jest.fn().mockReturnValue([
+    null,
+    {
+      data: {
+        indexPatterns: {
+          dataViewsServiceFactory: () => ({
+            get: jest.fn().mockReturnValue(mockPattern),
+          }),
+        },
+      },
+    },
+  ]);
+
+  beforeAll(() => {
+    getFieldsForWildcardMock.mockResolvedValue([]);
+    esClientSearchMock.mockResolvedValue({ hits: { total: { value: 123 } } });
+    esClientFieldCapsMock.mockResolvedValue({ indices: ['value'] });
+    IndexPatternsFetcher.prototype.getFieldsForWildcard = getFieldsForWildcardMock;
+  });
+
+  beforeEach(async () => {
+    const [
+      ,
+      {
+        data: { indexPatterns },
+      },
+    ] = await getStartServices();
+    IndexPatterns = indexPatterns;
+    getFieldsForWildcardMock.mockClear();
+    esClientSearchMock.mockClear();
+    esClientFieldCapsMock.mockClear();
+  });
+
+  afterAll(() => {
+    getFieldsForWildcardMock.mockRestore();
+  });
+  describe('with right privileges', () => {
+    it('should check index exists', async () => {
+      const indices = [eventsIndexPattern];
+      const request = {
+        indices,
+        onlyCheckIfIndicesExist: true,
+      };
+
+      const response = await requestEventFiltersFieldsSearch(
+        endpointAppContextService,
+        request,
+        deps,
+        beatFields,
+        IndexPatterns
+      );
+      expect(response.indexFields).toHaveLength(0);
+      expect(response.indicesExist).toEqual(indices);
+    });
+
+    it('should search index fields', async () => {
+      const indices = [eventsIndexPattern];
+      const request = {
+        indices,
+        onlyCheckIfIndicesExist: false,
+      };
+
+      const response = await requestEventFiltersFieldsSearch(
+        endpointAppContextService,
+        request,
+        deps,
+        beatFields,
+        IndexPatterns
+      );
+
+      expect(getFieldsForWildcardMock).toHaveBeenCalledWith({ pattern: indices[0] });
+
+      expect(response.indexFields).not.toHaveLength(0);
+      expect(response.indicesExist).toEqual(indices);
+    });
+
+    it('should throw when invalid index', async () => {
```

Comment:
consider adding a test for the case when there is more than one index.

Reply:
https://github.com/elastic/kibana/pull/145883/commits/7578fc13605307a95d0b8a13d069bb28e98b28b1